### PR TITLE
Add role to trainees who complete Full Roleplay

### PR DESCRIPTION
**Pull request by apackin**
#### What's this PR do?
Adds the CC1 Role for Trainees who complete assignment 'Full Roleplay' in Typeform.
This Role on CTL online allows them to start scheduling shifts.
#### How should this be manually tested?
Add a user to a Canvas course (I used myself)
Input passing grades for assignment 1-8
View user CTL Online Roles on https://online.crisistextline.org/user/${userID}/edit
  Make sure Role "Crisis Counselor 1" is not selected
Node www/app.js
Post to localhost:3000/canvas/typeformAssignment/Full Roleplay 
Use the following body with the email for Canvas account being used: 
 {
  "event_id": "UwVP9f9sGU",
  "event_type": "form_response",
  "form_response": {
    "form_id": "L5aBhG",
    "token": "875527530cbc8ec306774088db80887f",
    "submitted_at": "2016-08-12T14:49:14Z",
    "calculated": {
      "score": 2
    },
    "definition": {
      "id": "L5aBhG",
      "title": "",
      "fields": [
        {
          "id": "3",
          "title": "Question text: 3",
          "type": "textfield"
        },
        {
          "id": "2",
          "title": "Question Bool: 2>",
          "type": "yes-no"
        },
        {
          "id": "1",
          "title": "Please enter your email below",
          "type": "email"
        }
      ]
    },
    "answers": [
      {
        "type": "text",
        "text": "Sample Text Answer",
        "field": {
          "id": "3",
          "type": "short_text"
        }
      },
      {
        "type": "boolean",
        "boolean": true,
        "field": {
          "id": "2",
          "type": "yes_no"
        }
      },
      {
        "type": "email",
        "email": "apackin@gmail.com",
        "field": {
          "id": "1",
          "type": "email"
        }
      }
    ]
  }
}

Refresh https://online.crisistextline.org/user/${userID}/edit
  and the Role "Crisis Counselor 1" should be selected.

#### What are the relevant tickets?
https://admin.crisistextline.org/jira/secure/RapidBoard.jspa?rapidView=5&view=detail&selectedIssue=INT-396

### CODE REVIEW

File: ctlOnline.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+'use strict';
+
```
Comment (mbogor) [inline ctlOnline.js]: add comments before each function, just makes it easier

**Comment by mbogor** [comment]
👍

File: www/canvas/helpers.js [hunk 2]
```diff
@@ -37,5 +39,17 @@ function extractEmailAndFormatSubmission (formResponse) {
   return emailWithSubmission;
 }
 
+// Using this object because I believe we will have more triggers 
+// based on assignment names provided by the typeform webhook
```
Comment (tongxiang) [inline www/canvas/helpers.js]: Smart.

File: www/canvas/index.js [hunk 3]
```diff
@@ -52,7 +52,12 @@ router.post('/typeformAssignment/:assignment', function (req, res) {
     } else {
       return pSubmitAssign;
     }
-  }).then(() => res.send({message: `Successfully submitted ${assignment} with ${JSON.stringify(emailWithSubmission)}`}))
+  }).then(() => {
+    if (helpers.triggersBasedOnAssignment[assignment]) {
```
Comment (tongxiang) [inline www/canvas/index.js]: Also smart.

**Comment by tongxiang** [comment]
Well implemented. 👍

**Pull request merged**